components/CollectionCard.js

```diff
@@ -61,9 +91,44 @@ export default function CollectionCard({ flashcards, collection }) {
           <p>Correct Cards: {correctFlashcards.length}</p>
         </CollectionCardArticle>
       </StyledCardLink>
+
+      <DeleteButton
+        onClick={(event) => {
+          toggleConfirmation();
+        }}
+      >
+        delete
+      </DeleteButton>
+
       <StyledLink href={`/collections/${collection.id}/archive`}>
         Archive
       </StyledLink>
+
+      <StyledDialog open={showConfirmation}>
+        <h3>Do you really want to delete this collection?</h3>
+        <p>
+          🚨 This will also permanently delete all flashcards in this
+          collection.
+        </p>
+        <p>⚠️ This action cannot be undone!</p>
+        <>
+          <button
+            onClick={() => {
+              toggleConfirmation();
+            }}
```

Comment:
You could also get rid of the callback function, since you do not need an argument and only call one function inside it:

```suggestion
            onClick={toggleConfirmation}
```

And you could also add the type of the button to it.